cmd/object-api-utils.go

```diff
@@ -64,12 +69,13 @@ func IsValidBucketName(bucket string) bool {
 	if len(bucket) < 3 || len(bucket) > 63 {
 		return false
 	}
-	if bucket[0] == '.' || bucket[len(bucket)-1] == '.' {
```

Comment:
What i am saying is we can do rune compare without going into regex, regex we can use it for composite checks which are harder to achieve simply by rune but in lieu of readability.  I think this PR's approach is fine.